Create a new guide for upgrading LSP6 (#448)
lukso-network/docs: docs/guides/key-manager/upgrade-lsp6.md

---
sidebar_label: 'Upgrade LSP6 Key Manager'
sidebar_position: 4
describtion: 'This guide is useful for changing or upgrading the LSP6 Key Manager of a Universal Profile.'
---

import Tabs from '@theme/Tabs';
import TabItem from '@theme/TabItem';

# Upgrade the LSP6 Key Manager

:::info Requirements

You will need a Universal Profile that you can control via its [KeyManager](../../standards/universal-profile/lsp6-key-manager) to follow this guide.
If you don't have a Universal Profile yet, follow our previous guide [**Create a Universal Profile**](../universal-profile/create-profile.md) or look at the [_lsp-factory.js_](../../tools/lsp-factoryjs/deployment/universal-profile.md) docs.

:::

In this guide, we will learn how to upgrade the LSP6 Key Manager of your Universal Profile to the latest version available.

By the end of this guide, you will know how to:

- Deploy a new LSP6 Key Manager with the last updates.
- Upgrade your Key Manager by changing the owner of your UP from your old to your new Key Manager.

## Setup

```shell
npm install @lukso/lsp-smart-contracts web3
```

## Step 1 - Set up the constants and imports

> Create a javascript file and add the following imports on the top of the file.
> Create the following constants:
> 
> - `privateKey`: private key of a controller address, **MUST** have [**CHANGEOWNER**](../../standards/universal-profile/lsp6-key-manager.md#permissions) permission.

Reviewer: ```suggestion
Create a javascript file and add the following imports on the top of the file:

- `privateKey`: private key of a controller address, **MUST** have [**CHANGEOWNER**](../../standards/universal-profile/lsp6-key-manager.md#permissions) permission.
```

- `keyManagerAddress`: address of the current LSP6 Key Manager.
- `universalProfileAddress`: address of your Universal Profile.

```js
import LSP6KeyManager from '@lukso/lsp-smart-contracts/artifacts/LSP6KeyManager.json' assert { type: 'json' };

> import Web3 from 'web3';
> const web3 = new Web3('https://rpc.l14.lukso.network');

Reviewer: ```suggestion
import Web3 from 'web3';

const web3 = new Web3('https://rpc.l14.lukso.network');
```

const privateKey = '0x...';
const universalProfileAddress = '0x...';
const keyManagerAddress = '0x...';
```

## Step 2 - Initialize the controller account

> In order to send any transaction on the blockchain you need an account, in or case that account MUST have [**CHANGEOWNER**](../../standards/universal-profile/lsp6-key-manager.md#permissions) permission on the Universal Profile that will have its LSP6 Key Manager upgraded.

Reviewer: "an account, in or case that account" ??

Author: in our case

```js
const account = web3.eth.accounts.privateKeyToAccount(privateKey);
web3.eth.accounts.wallet.add(account);
```

## Step 3 - Initialize the old LSP6 Key Manager

In order to transfer ownership of your Universal Profile, you need to initialize your current LSP6 Key Manager.

```js

> const oldKeyManager = new web3.eth.Contract(
>   LSP6KeyManager.abi,
>   keyManagerAddress,
> );

Reviewer: ```suggestion
const oldKeyManager = new web3.eth.Contract(LSP6KeyManager.abi, keyManagerAddress);
```

const newKeyManager = await NewKeyManager.deploy({
  data: LSP6KeyManager.bytecode,
  arguments: [universalProfileAddress],
}).send({
  from: account.address,
  gas: 3000000,
  gasPrice: '1000000000',
});

console.log(newKeyManager._address);
```

## Step 5 - Upgrade your Key Manager

### Step 5.1 - Transfer Ownership to your new Key Manager

Create a payload for the [`transferOwnership(address)`](../../standards/smart-contracts/lsp14-ownable-2-step.md#transferownership) function and shift the ownership of your Universal Profile from your current LSP6 Key Manager.

```js
const transferOwnershipPayload =
  web3.eth.abi.encodeFunctionSignature('transferOwnership(address)') +
  web3.eth.abi.encodeParameter('address', newKeyManager._address).substring(2);

await oldKeyManager.methods['execute(bytes)'](transferOwnershipPayload).send({
  from: account.address,
  gas: 1000000,
  gasPrice: '1000000000',
});
```

### Step 5.2 - Accept Ownership from your new Key Manager

Create a payload for the [`claimOwnership()`](../../standards/smart-contracts/lsp14-ownable-2-step.md#acceptownership) function and take the ownership of your Universal Profile from your new LSP6 Key Manager.

```js

> const claimOwnershipPayload =
>   web3.eth.abi.encodeFunctionSignature('claimOwnership()');

Reviewer: ```suggestion
const claimOwnershipPayload =  web3.eth.abi.encodeFunctionSignature('claimOwnership()');
```

You can add //prettier-ignore above the codeblock.

It is ok to not put prettier on these examples as otherwise it will split super simple things into 1000 lines...
Having it in one line for the tutorial is fine

await newKeyManager.methods['execute(bytes)'](claimOwnershipPayload).send({
  from: account.address,
  gas: 1000000,
  gasPrice: '1000000000',
});
```

## The upgrade has been completed successfully.

## Final code

```javascript title="upgrade-lsp6.js"
import LSP6KeyManager from '@lukso/lsp-smart-contracts/artifacts/LSP6KeyManager.json' assert { type: 'json' };
import Web3 from 'web3';
const web3 = new Web3('https://rpc.l14.lukso.network');

> export const privateKey = '0x...';
> export const universalProfileAddress = '0x...';
> export const keyManagerAddress = '0x...';

Reviewer: If it is in one file, why do you need to export them?

```suggestion
const privateKey = '0x...';
const universalProfileAddress = '0x...';
const keyManagerAddress = '0x...';
```

    web3.eth.abi
      .encodeParameter('address', newKeyManager._address)
      .substring(2);

  await oldKeyManager.methods['execute(bytes)'](transferOwnershipPayload).send({
    from: account.address,
    gas: 1000000,
    gasPrice: '1000000000',
  });

  // Accept the ownership of your Universal Profile from the new LSP6 Key Manager
  const claimOwnershipPayload =
    web3.eth.abi.encodeFunctionSignature('claimOwnership()');

  await newKeyManager.methods['execute(bytes)'](claimOwnershipPayload).send({
    from: account.address,
    gas: 1000000,
    gasPrice: '1000000000',
  });
};

await upgradeLSP6();
```

## Test the new LSP6 Key Manager

We can now check the owner of the Universal Profile. If everything went through, the owner should be the address of the new LSP6 Key Manager.
Create the following file with the name `test-new-lsp6.js` and run:

```shell
node test-new-lsp6.js
```

```javascript title="test-new-lsp6.js"
import LSP0ERC725YAccount from '@lukso/lsp-smart-contracts/artifacts/LSP0ERC725YAccount.json' assert { type: 'json' };
import LSP6KeyManager from '@lukso/lsp-smart-contracts/artifacts/LSP6KeyManager.json' assert { type: 'json' };

> import { keyManagerAddress, universalProfileAddress } from './constants.js';

Reviewer: I think this constants.js file does not exist anymore

Maybe you need to put the const here too

import Web3 from 'web3';
const web3 = new Web3('https://rpc.l14.lukso.network');

const upgradeLSP6 = async () => {

>   const universalProfile = new web3.eth.Contract(
>     LSP0ERC725YAccount.abi,
>     universalProfileAddress,
>   );

Reviewer: ```suggestion
  const universalProfile = new web3.eth.Contract(LSP0ERC725YAccount.abi, universalProfileAddress);
```

see comment above for prettier

  const universalProfileOwner = await universalProfile.methods.owner().call();

>   console.log(
>     `The new owner of the Universal Profile is: ${universalProfileOwner}`,
>   );

Reviewer: ```suggestion
  console.log(`The new owner of the Universal Profile is: ${universalProfileOwner}`);
```

  console.log(`The old LSP6 Key Manager is at address: ${keyManagerAdderss}`);

  const keyManager = new web3.eth.Contract(
    LSP6KeyManager.abi,
    universalProfileOwner,
  );

  const keyManagerTarget = await keyManager.methods.target().call();

>   console.log(
>     `The address of the Universal Profile is: ${universalProfile._address}`,
>   );

Reviewer: ```suggestion
  console.log(`The address of the Universal Profile is: ${universalProfile._address}`);
```

:::

In this guide, we will learn how upgrade the LSP6 Key Manager of your Universal Profile to the latest version available.

By the end of this guide, you will know:

- How to deploy a new LSP6 Key Manager with the last updates.
- How to transfer ownership of your Universal Profile through the old LSP6 Key Manager.
- How to accept ownership of your Universal Profile through the new LSP6 Key Manager

## Setup

```shell
npm install @lukso/lsp-smart-contracts, web3
```

## Step 1 - Create `constants.js` file and save the needed constants

Create a `constants.js` file with the following constants

- PRIVATE_KEY = private key of a conroller address
- keyManagerAddress = address of the current LSP6 Key Manager
- universalProfileAddress = address of your Universal Profile
  This controller must have CHANGEOWNER permission

After this first step you can save & close `constants.js`.

```js
export const PRIVATE_KEY = '0x...';
export const universalProfileAddress = '0x...';
export const keyManagerAddress = '0x...';
```

> ## Step 2 - Create `upgrade-lsp6.js` and get the imports

Reviewer 2: Same thing about headings as above

```suggestion
## Step 2 - Integrate the Imports
```

Create a file with the name `upgarde-lsp6.js` and you should have the following imports on the top of the file.

```js
import LSP6KeyManager from '@lukso/lsp-smart-contracts/artifacts/LSP6KeyManager.json' assert { type: 'json' };
import {
  PRIVATE_KEY,
  keyManagerAddress,
  universalProfileAddress,

> } from './constants.js';
> import Web3 from 'web3';

Reviewer 2: Add spaces so it is more readable and users can differentiate between internal/external 
```suggestion
} from './constants.js';

import Web3 from 'web3';
```

const web3 = new Web3('https://rpc.l14.lukso.network');
```

## Step 3 - Initialize the controller account

> In order to use the controller account you need to initialize it using web3.js.

Reviewer 2: ```suggestion
In order to use the controller account you need to initialize it using the `web3.js` library.
```

```js
const account = web3.eth.accounts.privateKeyToAccount(PRIVATE_KEY);
web3.eth.accounts.wallet.add(account);
```

## Step 4 - Initialize the old LSP6 Key Manager

> In order to transfer ownership of your Universal Profile you need to initialize your current LSP6 Key Manager.

Reviewer 2: ```suggestion
In order to transfer ownership of your Universal Profile, you need to initialize your current LSP6 Key Manager.
```

```js
const oldKeyManager = new web3.eth.Contract(
  LSP6KeyManager.abi,
  keyManagerAddress,
);
```

## Step 5 - Deploy the new LSP6 Key Manager

Deploy a new LSP6 Key Manager with the latest updates.

```js
const NewKeyManager = new web3.eth.Contract(LSP6KeyManager.abi);
const newKeyManager = await NewKeyManager.deploy({
  data: LSP6KeyManager.bytecode,
  arguments: [universalProfileAddress],
}).send({
  from: account.address,
  gas: 3000000,
  gasPrice: '1000000000',
});

console.log(newKeyManager._address);
```

## Step 6 - Transfer ownership of the Universal Profile

> Create a payload for `transferOwnership(address)` and tarnsfer the ownership of your Universal Profile from your current LSP6 Key Manager.

Reviewer 2: Maybe just a minor change, but it was kinda hard to read to have "transfer ownership" right after each other
```suggestion
Create a payload for the `transferOwnership(address)` function and shift the ownership of your Universal Profile from your current LSP6 Key Manager.
```

```js
const transferOwnershipPayload =
  web3.eth.abi.encodeFunctionSignature('transferOwnership(address)') +
  web3.eth.abi.encodeParameter('address', newKeyManager._address).substring(2);

await oldKeyManager.methods['execute(bytes)'](transferOwnershipPayload).send({
  from: account.address,
  gas: 1000000,
  gasPrice: '1000000000',
});
```

## Step 7 - Claim ownership of the Universal Profile

> Create a payload for `claimOwnership()` and claim the ownership of your Universal Profile from your new LSP6 Key Manager.

Reviewer 2: ```suggestion
Create a payload for the `claimOwnership()` function and take the ownership of your Universal Profile from your new LSP6 Key Manager.
```

```js
const claimOwnershipPayload =
  web3.eth.abi.encodeFunctionSignature('claimOwnership()');

await newKeyManager.methods['execute(bytes)'](claimOwnershipPayload).send({
  from: account.address,
  gas: 1000000,
  gasPrice: '1000000000',
});
```

## Final code

```javascript title="constants.js"
export const PRIVATE_KEY = '0x...';
export const universalProfileAddress = '0x...';
export const keyManagerAddress = '0x...';
```

```javascript title="upgrade-lsp6.js"
import LSP6KeyManager from '@lukso/lsp-smart-contracts/artifacts/LSP6KeyManager.json' assert { type: 'json' };
import {
  PRIVATE_KEY,
  keyManagerAddress,
  universalProfileAddress,

> } from './constants.js';
> import Web3 from 'web3';

Reviewer 2: ```suggestion
} from './constants.js';

import Web3 from 'web3';
```

const web3 = new Web3('https://rpc.l14.lukso.network');

const upgradeLSP6 = async () => {
  // Initialize the controller account
  const account = web3.eth.accounts.privateKeyToAccount(PRIVATE_KEY);
  web3.eth.accounts.wallet.add(account);

  // Initialize your current LSP6 Key Manager
  const oldKeyManager = new web3.eth.Contract(
    LSP6KeyManager.abi,
    keyManagerAddress,
  );

>   // Deploy a new LSP6 Key Manager

Reviewer 2: @CJ42 whats the right convention? Is it LSP6 Key Manager or LSP6 KeyManager? Just worried myself, nothing to judge here.

Author: I saw in other guides it was separate so I kept it that way

Reviewer 2: Ok, great- so I know how to write standard names from now on 👍🏻

  const NewKeyManager = new web3.eth.Contract(LSP6KeyManager.abi);
  const newKeyManager = await NewKeyManager.deploy({
    data: LSP6KeyManager.bytecode,
    arguments: [universalProfileAddress],
  }).send({
    from: account.address,
    gas: 3000000,
    gasPrice: '1000000000',
  });

  console.log(newKeyManager._address);

  // Transfer the ownership of your Universal Profile from the current LSP6 Key Manager to a new LSP6 Key Manager
  const transferOwnershipPayload =
    web3.eth.abi.encodeFunctionSignature('transferOwnership(address)') +
    web3.eth.abi
      .encodeParameter('address', newKeyManager._address)
      .substring(2);

  await oldKeyManager.methods['execute(bytes)'](transferOwnershipPayload).send({
    from: account.address,
    gas: 1000000,
    gasPrice: '1000000000',
  });

  // Accept the ownership of your Universal Profile from the new LSP6 Key Manager
  const claimOwnershipPayload =
    web3.eth.abi.encodeFunctionSignature('claimOwnership()');

  await newKeyManager.methods['execute(bytes)'](claimOwnershipPayload).send({
    from: account.address,
    gas: 1000000,
    gasPrice: '1000000000',
  });
};

await upgradeLSP6();
```

> ## Testing the new LSP6 Key Manager

Reviewer 2: As you always used simple present in other headings, i would keep this one in line too- removing the present progressive. I would suggest to always have a look at the headings on the right side and see if they align on the page- kinda hard to do this from plain code. It also helps with adapting the length of headings, so they look decent all together.
```suggestion
## Test the new LSP6 Key Manager
```

> We can now check the owner of the Universal Profile and it should be the new LSP6 Key Manager.

Reviewer 2: ```suggestion
We can now check the owner of the Universal Profile. If everything went through, the owner should be the address of the new LSP6 Key Manager.
```

---
sidebar_label: 'Upgrade LSP6 Key Manager'
sidebar_position: 4

> ---

Reviewer: Maybe you can add a page description?

https://docusaurus.io/docs/markdown-features/head-metadata#markdown-page-description

As the first full line here may not be super relevant.

Keep it under 160 chars.

This can also be applied to other pages to improve SEO.

import Tabs from '@theme/Tabs';
import TabItem from '@theme/TabItem';

# Upgrade the LSP6 Key Manager

:::info Requirements

> You will need a Universal Profile that you can control via its KeyManager to follow this guide.

Reviewer 3: this callout can be deleted imo

Author: I feel like it is a requirement in order to follow the guide 🤔

> You will need a Universal Profile that you can control via its KeyManager to follow this guide.

Reviewer: What about adding some internal links?

```suggestion
You will need a Universal Profile that you can control via its [KeyManager](../../../standards/universal-profile/lsp6-key-manager) to follow this guide.
```

(Please check the number of `../` i did it from GitHub im not sure if it is correct)

If you don't have a Universal Profile yet, follow our previous guide [**Create a Universal Profile**](../universal-profile/create-profile.md) or look at the [_lsp-factory.js_](../../tools/lsp-factoryjs/deployment/universal-profile.md) docs.

:::

In this guide, we will learn how to upgrade the LSP6 Key Manager of your Universal Profile to the latest version available.

By the end of this guide, you will know how to:

- Deploy a new LSP6 Key Manager with the last updates.

> - Transfer ownership of your Universal Profile through the old LSP6 Key Manager.

Reviewer 3: Maybe we can merge the last two bullet points in one by saying:

- upgrade your Key Manager by changing the owner of your UP from your old to your new Key Manager.

wdyt?

- Accept ownership of your Universal Profile through the new LSP6 Key Manager

## Setup

```shell
npm install @lukso/lsp-smart-contracts web3
```

> ## Step 1 - Set up constants

Reviewer 3: Maybe we can integrate Step 1 and 2 together?

It might be easier to just put three constant values in the same single file.

So the code snippet below would look like this:

```js
import LSP6KeyManager from '@lukso/lsp-smart-contracts/artifacts/LSP6KeyManager.json' assert { type: 'json' };
import Web3 from 'web3';
const web3 = new Web3('https://rpc.l14.lukso.network');

const privateKey = '0x...';
const universalProfileAddress = '0x...';
const keyManagerAddress = '0x...';
```

Create a `constants.js` file with the following constants

- `privateKey`: private key of a controller address
- `keyManagerAddress`: address of the current LSP6 Key Manager
- `universalProfileAddress`: address of your Universal Profile
  This controller must have the CHANGEOWNER permission

After this first step you can save & close `constants.js`.

```js
export const privateKey = '0x...';
export const universalProfileAddress = '0x...';
export const keyManagerAddress = '0x...';

> ```

Reviewer: I don't think it is necessary to split things in multiple files.

These guides are about showing how to do things, it is ok to duplicate and to not have a clean code.

By putting these 3 lines, without export, on the main file, you make the tutorial easier to read.

Also the final code will be easier as users can simply "copy/paste" and they don't have to create another file just to save 3 variables in them.

> ## Step 2 - Integrate the Imports

Reviewer: What does "integrate the imports" mean?

Author: Not a good name, will remove it when adding the constants to the main file

Create a file with the name `upgarde-lsp6.js` and you should have the following imports on the top of the file.

```js
import LSP6KeyManager from '@lukso/lsp-smart-contracts/artifacts/LSP6KeyManager.json' assert { type: 'json' };
import {
  privateKey,
  keyManagerAddress,
  universalProfileAddress,
} from './constants.js';

import Web3 from 'web3';
const web3 = new Web3('https://rpc.l14.lukso.network');
```

## Step 3 - Initialize the controller account

> In order to use the controller account you need to initialize it using the `web3.js` library.

Reviewer: the init is done above on the "new Web3(" line.

Maybe you can say "you need to add an account" rather than "you need to initialise web3

```js
const account = web3.eth.accounts.privateKeyToAccount(privateKey);
web3.eth.accounts.wallet.add(account);
```

> ## Step 4 - Initialize the old LSP6 Key Manager

Reviewer 3: I would suggest that we move this step in the Step 6 below.

The initial step is to deploy the new Key Manager on the network. So this contract is now ready to be used and linked to the UP.

So after deploying the new Key Manager contract, we could make two sub-sections under step 6.

```
## Step 6 - Upgrade your Key Manager

### Step 6.1 - Transfer Ownership to your new Key Manager

// We create an instance of the old Key Manager (using this code snippet below). We call it with the `transferOwnership(...)` payload and the address of the new Key Manager we deployed in the previous step.

### Step 6.2 - Accept Ownership from your new Key Manager

// then we create an other instance of the new Key Manager, same passing the address of the new Key Manager we deployed in the previous step. And from this new Key Manager, we send the `acceptOwnership(...)` payload.

## The upgrade has been completed successfully.
```

In order to transfer ownership of your Universal Profile, you need to initialize your current LSP6 Key Manager.

```js
const oldKeyManager = new web3.eth.Contract(
  LSP6KeyManager.abi,
  keyManagerAddress,
);
```

## Step 5 - Deploy the new LSP6 Key Manager

Deploy a new LSP6 Key Manager with the latest updates.

```js
const NewKeyManager = new web3.eth.Contract(LSP6KeyManager.abi);
const newKeyManager = await NewKeyManager.deploy({
  data: LSP6KeyManager.bytecode,
  arguments: [universalProfileAddress],
}).send({
  from: account.address,
  gas: 3000000,
  gasPrice: '1000000000',
});

console.log(newKeyManager._address);
```

## Step 6 - Transfer ownership of the Universal Profile

Create a payload for the `transferOwnership(address)` function and shift the ownership of your Universal Profile from your current LSP6 Key Manager.

```js
const transferOwnershipPayload =

>   web3.eth.abi.encodeFunctionSignature('transferOwnership(address)') +

Reviewer 3: maybe we can use the `contract.methodName(...).encodeABI()` function here?

```js
const transferOwnershipPayload = await web3.methods.transferOwnership(newKeyManager._address).encodeABI()
```

Author: you don't have the up contract with that method

  const transferOwnershipPayload =
    web3.eth.abi.encodeFunctionSignature('transferOwnership(address)') +
    web3.eth.abi
      .encodeParameter('address', newKeyManager._address)
      .substring(2);

  await oldKeyManager.methods['execute(bytes)'](transferOwnershipPayload).send({
    from: account.address,
    gas: 1000000,
    gasPrice: '1000000000',
  });

  // Accept the ownership of your Universal Profile from the new LSP6 Key Manager
  const claimOwnershipPayload =
    web3.eth.abi.encodeFunctionSignature('claimOwnership()');

  await newKeyManager.methods['execute(bytes)'](claimOwnershipPayload).send({
    from: account.address,
    gas: 1000000,
    gasPrice: '1000000000',
  });
};

await upgradeLSP6();
```

## Test the new LSP6 Key Manager

We can now check the owner of the Universal Profile. If everything went through, the owner should be the address of the new LSP6 Key Manager.
Create the following file with the name `test-new-lsp6.js` and run:

```shell
node test-new-lsp6.js
```

```javascript title="test-new-lsp6.js"
import LSP0ERC725YAccount from '@lukso/lsp-smart-contracts/artifacts/LSP0ERC725YAccount.json' assert { type: 'json' };

> import LSP6KeyManager from '@lukso/lsp-smart-contracts/artifacts/LSP6KeyManager.json' assert { type: 'json' };

Reviewer 3: why is there an `assert` here at the end of the import?

Author: because you get an error when importing json

Reviewer 3: @Hugoo any ideas on this?

import { keyManagerAddress, universalProfileAddress } from './constants.js';
import Web3 from 'web3';
const web3 = new Web3('https://rpc.l14.lukso.network');

const upgradeLSP6 = async () => {

>   const unviersalProfile = new web3.eth.Contract(

Reviewer 3: ```suggestion
  const universalProfile = new web3.eth.Contract(
```

    LSP0ERC725YAccount.abi,
    universalProfileAddress,
  );

  const universalProfileOwner = await universalProfile.methods.owner().call();
  console.log(
    `The new owner of the Universal Profile is: ${universalProfileOwner}`,
  );
  console.log(`The old LSP6 Key Manager is at address: ${keyManagerAdderss}`);

  const keyManager = new web3.eth.Contract(
    LSP6KeyManager.abi,
    universalProfileOwner,
  );

  const keyManagerTarget = await keyManager.methods.target().call();

  console.log(

>     `The address of the Universal Profile is: ${unviersalProfile._address}`,

Reviewer 3: ```suggestion
    `The address of the Universal Profile is: ${universalProfile._address}`,
```

:::info Requirements

> You will need a Universal Profile that you can control via its [KeyManager](../../standards/universal-profile/lsp6-key-manager) to follow this guide.

Reviewer: .md or i think the link might be broken - cf one of the latest closed PR

## Step 1 - Set up the constants and imports

> Create a javascript file and add the following imports on the top of the file:

Reviewer: JavaScript

In order to transfer ownership of your Universal Profile, you need to initialize your current LSP6 Key Manager.

```js

> //prettier-ignore

Reviewer: i think if you put it here it will show in the page. This is what i meant:

https://github.com/lukso-network/docs/blob/2b56b521020f1a48468599bfbf4874545aad8d62/docs/guides/browser-extension/sign-in-with-ethereum.md?plain=1#L63
